assets/src/skate.d.ts

```diff
@@ -79,6 +79,8 @@ export interface Vehicle {
 
 export type VehicleId = string
 
+export type VehicleAdherenceStatus = ScheduleAdherenceStatus | "off-course"
```

Comment:
My initial reaction is that this feels like conflating two things that aren't really the same thing (we are just using in many of the same places at the moment). I think about headway-based routes for example. Schedule adherence will not be used when we are drawing those, but we'll still want to draw a black triangle for invalid data.

Reply:
Mm. I see.

Even if the headway view will draw invalid and on-time statuses differently, I think they might still belong together in the data. The statuses don't overlap: a vehicle is exactly one of `on-time`, `late`, `early` or `off-course`, so putting them in the same field/type is a better description of the vehicle and reduces illegal states. The headway view code would have to have an `if (status === "off-course") {black triangle} else {grey triangle}`, but I think that's a fine way for view logic to look at the data, and we basically have that already where the `ladderVehicle` decides whether to draw the triangle at the scheduled or realtime location.

Alternatively, if they really are different: Since `VehicleAdherenceStatus` is just used for the `LadderVehicle` and vehicle icon, maybe `off-course` should be a view concept, and this definition could be moved out of this file (which is just for data), and into a file that works more with view stuff. Maybe `vehicleStatus`?

Reply:
Ok